htsohm/htsohm.py

```diff
@@ -455,24 +455,40 @@ def worker_run_loop(run_id):
                 session.add(material)
 
             if config['mutation_strength_method'] != 'flat':
-                if (
-                    material.generation_index == config['children_per_generation'] - 1
-                    and gen > 0
-                    ):
-                    parent_ids = get_all_parent_ids(run_id, gen)
-    
-                    print_block('CALCULATING MUTATION STRENGTHS')
-                    ms_bins = []
-                    for parent_id in parent_ids:
-                        parent_bin = session.query(Material).get(parent_id).bin
-                        if parent_bin not in ms_bins:
-                            print(
-                                    (
-                                        'Calculating bin-mutation-strength for bin : {0}'
-                                    ).format(parent_bin)
-                                )
-                            calculate_mutation_strength(run_id, gen + 1, parent_bin)
-                        ms_bins.append(parent_bin)
+                if material.generation_index == config['children_per_generation'] - 1 and gen > 0:
+                # standard calculation of mutation strengths for all accessed bins
+                    if gen % config['annealing_frequency'] != 0:
```

Comment:
generally, when I don't do this (for example, with SJS), I regret it because somebody sets the number to something REALLY high, then gets surprised when it happens then. So probably add a config setting and you can still do: if config['annealing_on'] and (gen % config['annealing_frequency'] != 0):
Not too much more complicated, and it is explicit, rather than implied.